bgpd/bgp_flowspec.c

```diff
@@ -124,7 +117,11 @@ int bgp_nlri_parse_flowspec(struct peer *peer, struct attr *attr,
 			return BGP_NLRI_PARSE_ERROR_PACKET_OVERFLOW;
 
 		psize = *pnt++;
-
+		if (psize >= FLOWSPEC_NLRI_SIZELIMIT) {
+			psize &= 0x0f;
```

Comment:
1. Why is `psize` an int? The RFC says it's at most 2 bytes.
2. This could just be `psize = (0x0f00 & *pnt++)`; but even then,
3. I'm pretty sure this is just straight up incorrect; from the RFC:
```
   If the NLRI length value is smaller than 240 (0xf0 hex), the length
   field can be encoded as a single octet.  Otherwise, it is encoded as
   an extended-length 2-octet value in which the most significant nibble
   of the first byte is all ones.
...
   Values above 240 are encoded using 3 hex digits (0xfnnn)
```
The most significant nibble as a result of this code is all 0's, the next nibble is all 1s, and the extended-length value is just truncated into the bottom byte because this code still only reads 1 byte of length, so all that's been done here is make this code handle lengths up to 255 instead of the maximum value of 4095:
```
The highest value that can be represented with this encoding is 4095
```